app/components/ui/molecules/PeoplePicker.js

```diff
@@ -145,11 +147,57 @@ class PeoplePicker extends React.Component {
     this.setState({ searchValue })
   }
 
-  filterPeople(people) {
-    const searchValue = this.state.searchValue.toLowerCase()
-    return people.filter(person =>
-      person.name.toLowerCase().includes(searchValue),
-    )
+  filterPeople = (people, searchValue, field) => {
+    if (!searchValue) return people
+
+    const inputValue = searchValue.trim().toLowerCase()
+    if (!inputValue) return people
+
+    if (inputValue.split(' ').length > 1) {
+      const matches = people.filter(person =>
+        person[field].toLowerCase().includes(inputValue),
+      )
+      return matches.map(elem => {
+        const clonedElem = cloneDeep(elem)
+        clonedElem.matchIndex = elem[field].toLowerCase().indexOf(inputValue)
+        return clonedElem
+      })
+    }
+    return this.filterPeopleSingle(people, inputValue, field)
+  }
+
+  filterPeopleSingle = (people, inputValue, field) => {
+    let maxWords = 0
+    people.forEach(person => {
+      const numWords = person[field].match(/[^ ]+/g).length
```

Comment:
What does this regex match?

Reply:
It matches all the words. I haven't done enough testing to be fair, if you know an edge case where this is not working feel free to say :)